pkg/kapp/app/recorded_app.go

```diff
@@ -115,7 +115,7 @@ func (a *RecordedApp) UpdateUsedGVsAndGKs(gvs []schema.GroupVersion, gks []schem
 	})
 }
 
-func (a *RecordedApp) CreateOrUpdate(labels map[string]string) error {
+func (a *RecordedApp) CreateOrUpdate(labels map[string]string, isDiffRun bool) error {
```

Comment:
@sethiyash Can somebody try to use `diffRun` while doing migration. If yes, are all those cases handled ?

Reply:
tested with different scenario.

Reply:
Resource| KAPP_FQ_CONFIGMAP_NAMES | diff-run | create  | update  | delete    | CM created |
| --------- | ---------------- | -------- | ------- | ------- | --------- | ------- |
| new       | False            | no       | Success | Success | Success   | yes     |
| new       | False            | yes      | Success | Success | Success | no      |
| new       | True         | no       |   Success | Success        |   Success        |   yes      |
| new       |  True         | yes|  Success |   Success      |   Success        |  No       |
| existing  | True     |    yes |  Success  |   Success     |   Success        |     No    |
| existing       | True             | no      |   Success      |     Success    |  Success         |      yes   |